packages/hooks/src/hooks/useWallet.ts

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Web3Context } from '../Provider';
+
+/**
+ * @dev Hook to get the current web3 context inlcuding information about the connection and other helper methods.
+ */
+export function useWallet() {
+  const context = React.useContext(Web3Context);
+  const [ens, setEns] = React.useState<string>();
+
+  React.useEffect(() => {
+    if (context) {
+      const { userAddress, provider } = context;
+      if (userAddress && provider) {
+        provider.lookupAddress(userAddress).then((address) => {
+          setEns(address as string);
```

Comment:
do we need to handle an error case where the address isn't found?

Reply:
No. In that case, the ens will simply be null which is fine.